lib/ansible/module_utils/service.py

```diff
@@ -207,17 +207,19 @@ def daemonize(module, cmd):
         p = subprocess.Popen(run_cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=lambda: os.close(pipe[1]))
         fds = [p.stdout, p.stderr]
 
-        # loop reading output till its done
+        # loop reading output till it is done
         output = {p.stdout: b(""), p.stderr: b("")}
         while fds:
             rfd, wfd, efd = select.select(fds, [], fds, 1)
-            if (rfd + wfd + efd) or p.poll():
+            if (rfd + wfd + efd) or p.poll() is None:
                 for out in list(fds):
                     if out in rfd:
                         data = os.read(out.fileno(), chunk)
                         if not data:
                             fds.remove(out)
-                    output[out] += b(data)
+                        output[out] += data
```

Comment:
in the end i think this is how this part should look
```
if data:
   output[out] += b(data)
else:
   fds.remove(out)
```

Reply:
@bcoca this will result in `'bytes' object has no attribute 'encode'`

Reply:
true, should use to_bytes() instead

Reply:
or just test for string vs bytestring .. mostly this is py2 compat code

Reply:
@bcoca, I see. In this case it probably makes sense to open a corrective PR to replace that block as you suggested?

Reply:
@bcoca #81584 ready for your review.